frontend/pages/hosts/details/modals/MDMStatusModal/MDMStatusModal.tsx

```
@@ -471,7 +474,14 @@ const MDMStatusModal = ({
   return (
     <Modal title="MDM status" className={baseClass} onExit={onExit}>
       {renderMDMStatus()}
-      {isPremiumTier && isAppleDevice && renderProfileAssignment()}
+      {isPremiumTier &&
+        isAppleDevice &&
+        // Only render the modal if this host has an actual host_dep_assignment entry,
+        // in which case we expect there to be data to render. While loading, keep
+        // the section visible so renderProfileAssignmentList can show its spinner.
+        (isLoadingDepAssignment ||
+          depAssignmentData?.host_dep_assignment) &&
```

ESLint

[error] Delete `⏎·········`

(prettier/prettier)